src/file/read-file.js

```diff
@@ -0,0 +1,20 @@
+var fs = require('fs');
+var Promise = require('bluebird');
+var readTextFile = require('../read-text-file');
+var sourceMap = require('./source-map');
+
+function read(filePath) {
+
+    return new Promise(function(resolve, reject) {
+
+        readTextFile(filePath, function(code) {
+
+            resolve(sourceMap.extract(code));

```

Comment:
When reading already compiled files off disk, this extracts the source map comment from the file and returns an object with the code and the decoded source map, which is what gets returned for files we *are* compiling rather than reading off disk.